src/Kernel/Magic/ViewMagic.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Jupyter.Core.Protocol;
+using Microsoft.Quantum.IQSharp.Core.ExecutionPathTracer;
+using Microsoft.Quantum.IQSharp.Jupyter;
+using Microsoft.Quantum.Simulation.Simulators;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Quantum.IQSharp.Kernel
+{
+    /// <summary>
+    ///      Contains the JSON representation of the <see cref="ExecutionPath"/>
+    ///      and the ID of the HTML div that will contain the visualization.
+    /// </summary>
+    public class ExecutionPathVisualizerContent : MessageContent
+    {
+        /// <summary>
+        ///     Initializes <see cref="ExecutionPathVisualizerContent"/> with the
+        ///     given <see cref="ExecutionPath"/> (as a <see cref="JToken"/>) and HTML div ID.
+        /// </summary>
+        public ExecutionPathVisualizerContent(JToken executionPath, string id)
+        {
+            this.ExecutionPath = executionPath;
+            this.Id = id;
+        }
+
+        /// <summary>
+        ///     The <see cref="ExecutionPath"/> (as a <see cref="JToken"/>) to be rendered.
+        /// </summary>
+        [JsonProperty("executionPath")]
+        public JToken ExecutionPath { get; }
+
+        /// <summary>
+        ///     ID of the HTML div that will contain the visualization.
+        /// </summary>
+        [JsonProperty("id")]
+        public string Id { get; }
+    }
+
+    /// <summary>
+    ///     A magic command that can be used to visualize the execution
+    ///     paths of operations and functions.
+    /// </summary>
+    public class ViewMagic : AbstractMagic
+    {
+        private const string ParameterNameOperationName = "__operationName__";
+
+        /// <summary>
+        ///     Constructs a new magic command given a resolver used to find
+        ///     operations and functions, and a configuration source used to set
+        ///     configuration options.
+        /// </summary>
+        public ViewMagic(ISymbolResolver resolver, IConfigurationSource configurationSource) : base(
+            "view",
+            new Documentation
+            {
+                Summary = "Outputs an HTML-based visualization of the execution path of a given operation."
+            })
```

Comment:
Before merging this PR, it would be great to expand this documentation a little bit to include markdown for `Description` and `Examples`, similar to other magic commands. Probably you can look at the [`%simulate` documentation](https://github.com/microsoft/iqsharp/blob/337e9e717aa83c1cb9ad420ea4d1e1183776475f/src/Kernel/Magic/Simulate.cs#L28-L63) as a good starting point, since you have the same parameter requirements for this magic command.

This info gets automatically published to the public API docs thanks to some work @cgranade did recently, and it is also viewable inside Jupyter Notebook by appending a `?` to the magic command name, e.g. `%view?`.